src/webapp/src/App.js

```diff
@@ -1,48 +1,44 @@
-import React, { Component } from 'react';
-import 'typeface-roboto/index.css'
+import React, { useEffect, useState } from "react";
+import "typeface-roboto/index.css";
 // import 'material-design-icons/iconfont/material-icons.css'
-import './App.css';
-import { Header } from './components/Header'
-import Footer from './components/Footer'
-import { Token } from './components/Token'
-import { ReposList } from './components/ReposList'
-import EulaModal from './components/EulaModal'
-import { ipcRenderer } from 'electron';
-import './App.scss';
+import "./App.css";
+import { Header } from "./components/Header";
+import Footer from "./components/Footer";
+import { Token } from "./components/Token";
+import { ReposList } from "./components/ReposList";
+import { EulaModal } from "./components/EulaModal";
+import { ipcRenderer } from "electron";
+import "./App.scss";
 
-class App extends Component {
-  constructor(props) {
-    super(props);
-    this.state = {
-      loading: true
-    }
+export const App = ({ ...props }) => {
+  const [loading, setLoading] = useState(true);
+  const [hasSignedEula, setHasSignedEula] = useState(ipcRenderer.sendSync("has-signed-eula"));
+
+  useEffect(() => {
     ipcRenderer.on("indexer-worker-id", (e, id) => {
```

Comment:
Please add small comment saying `"indexer-worker-id"` is only called after `"app-window-up"`